zipline/utils/calendars/trading_calendar.py

```diff
@@ -677,8 +677,13 @@ def days_at_time(days, t, tz, day_offset=0):
     day_offset : int
         The number of days we want to offset @days by
     """
-    days = DatetimeIndex(days).tz_localize(None).tz_localize(tz)
+
+    # Offset days in UTC to avoid discontinuities due to DST, before
+    # localizing to tz.
+    days = DatetimeIndex(days).tz_localize(None).tz_localize('UTC')
```

Comment:
Is that a bug in this function, or are we working around a pandas bug?  I don't understand how the changes here fix that problem.